### PR TITLE
Jukes cantor likelihood

### PR DESCRIPTION
**Pull request by willdumm**
This PR provides an implementation of Jukes Cantor likelihood on histories, with branch lengths inferred to match the expected number of mutations on each branch to the observed number of mutations on that branch.

Also included:
* implements `__contains__` method so that `history in dag` returns whether `history` is among the histories in `dag`
* provides a way to estimate the number of bifurcating histories in the DAG. This was intended to provide a new way of computing node support by summing over all bifurcating trees, but those plans have been scrapped.

### CODE REVIEW

File: tests/test_factory.py [hunk 1]
```diff
@@ -913,6 +805,19 @@ def test_node_support():
         assert is_close(exp(nd[node]), od[node], tol=0.0001)
 
 
+def test_conditional_annotation_nilpotent():
```
Comment (marybarker) [inline tests/test_factory.py:808]: This is a super fun test!

**Review by marybarker — APPROVED** [review]
Everything looks great! It might be good to incorporate a test that asserts likelihood implementations satisfy basic sanity tests, both to insure that and to formalize their function calls and so forth in case any calls change in future, but I don't think that is a necessary addition at this point

**Comment by matsen** [comment]
> but those plans have been scrapped

Can you remind me-- this is because it just didn't work well?

**Comment by willdumm** [comment]
For now I've abandoned the idea because there doesn't seem to be a good way of counting bifurcating resolutions correctly, since the new nodes that would result from resolving a multifurcating node may already be in the DAG. I think there's something more involved that could be done that would involve changing the data structure slightly...

**Pull request merged**